js/src/lib/components/InferenceWidget/shared/WidgetTextarea/WidgetTextarea.svelte

```diff
@@ -1,29 +1,34 @@
 <script>
+	import { onMount } from "svelte";
+
 	import WidgetLabel from "../WidgetLabel/WidgetLabel.svelte";
 
 	export let label: string = "";
 	export let placeholder: string = "Your sentence here...";
 	export let value: string;
 
-	// hack to handle FireFox contenteditable bug
-	let innterHTML: string;
 	let spanEl: HTMLSpanElement;
-	const REGEX_SPAN = /<span .+>(.*)<\/span>/gms;
-	$: {
-		if (spanEl && innterHTML && REGEX_SPAN.test(innterHTML)) {
-			innterHTML = innterHTML.replace(REGEX_SPAN, (_, txt) => {
-				return txt;
-			});
-			spanEl.blur();
-		}
+
+	function insertTextAtCaret(text: string) {
+		const sel = window.getSelection();
+		const range = sel.getRangeAt(0);
+		range.insertNode(document.createTextNode(text));
+		range.collapse(false);
 	}
+
+	onMount(() => {
+		spanEl.addEventListener("paste", (e) => {
```

Comment:
You should remove the eventListener on unMount :)

Reply:
And to achieve this, also probably make the arrow function a named handling function in the body of the component

Reply:
Great suggestion!